app/scripts/metamask-controller.js

```diff
@@ -2639,7 +2639,37 @@ export default class MetamaskController extends EventEmitter {
             this.controllerMessenger,
             'SnapController:clearSnapState',
           ),
-          getMnemonic: this.getPrimaryKeyringMnemonic.bind(this),
+          getMnemonic: async (source) => {
+            if (!source) {
+              return this.getPrimaryKeyringMnemonic();
+            }
+
+            try {
+              const { type, mnemonic } = await this.controllerMessenger.call(
+                'KeyringController:withKeyring',
+                {
+                  id: source,
+                },
+                async (keyring) => ({
+                  type: keyring.type,
+                  mnemonic: keyring.mnemonic,
+                }),
+              );
+
+              if (type !== KeyringTypes.hd || !mnemonic) {
```

Comment:
This doesn't match the one we use in `getPrimaryKeyringMnemonic`: https://github.com/MetaMask/metamask-extension/blob/8e757080706ca4774a0110ed5688943609fe2aeb/app/scripts/metamask-controller.js#L4688

Reply:
They refer to the same string, but `KeyringTypes` is what's used in the `keyring-controller`. I think it makes more sense to use that?

Reply:
Ah okay, sure 👍